Refactor folder structure to increase component reuse

andre1050 opened the pull request:
This is a long PR but nothing has changed in terms of component output. We're still rendering the same emails and variants using the same components. This is mostly moving components around to help reduce coupling between emails and the components they use.

This is a big step towards the goal of truly reusable components whereby each card design can be used in whatever email we need it on, without code duplication. But we're not there yet as more effort is needed inside the 'Opinion' folder to share components across both variants.

This PR is roughly half way there but we should merge sooner rather than later as there's work happening in parallel on `master` in files that have now been moved around.

nicl commented on src/api.ts:
```diff
@@ -147,6 +145,12 @@ interface SEOData {
     webTitle: string;
 }
 
+export enum Fronts {
```
Could this live just in `Email.tsx` as it is only used there? It's not really related to the api model itself/rest of this file.

andre1050 replied:
Makes sense, I've moved it to `Email.tsx`

andre1050 commented:
Thanks @nicl for the comments and feedback! You make very good points about the coupling between the layouts and the API. I'll be making some changes/updates to this today based on your comments and our discussion yesterday.

andre1050 commented:
hey @nicl I've made some further changes following your feedback.

- Removed the needless hardcoded assumptions on the order of `collections` within each email, by ensuring we always iterate through the collections and use the `getDesignType` function to determine which particular `Collection` component can handle that particular design type.

- Restored this iteration through collections in the top-level (generic) `Collections` and `Collection` components, but seeing as these are generic and often used as a fallback, no `getDesignType` is used and instead each collection will get the same treatment (headline, multiline, grid of 6 items...); this means we can default to handle any type of email in this way without any code changes.

- Added a `fronts` folder at the top level, which is where each specific type of email lives (e.g. `film-today`, `media-briefing`, etc.); at the root of each of these, we determine which variant to render, then iterate through the collections and render them using a series of variant-specific Collection components; the `Collections` component at the root of each variant is identical but actually imports components from a subfolder that's variant-aware (i.e. even though the files are exactly identical, the imports which they render are totally different). Open to discussing alternatives to this.
